pkg/providers/version/version.go

```diff
@@ -53,10 +52,12 @@ type Provider interface {
 // DefaultProvider get the APIServer version. This will be initialized at start up and allows karpenter to have an understanding of the cluster version
 // for decision making. The version is cached to help reduce the amount of calls made to the API Server
 type DefaultProvider struct {
-	cache               *cache.Cache
-	cm                  *pretty.ChangeMonitor
-	kubernetesInterface kubernetes.Interface
-	eksapi              sdk.EKSAPI
+	cache                 *cache.Cache
+	cm                    *pretty.ChangeMonitor
+	kubernetesInterface   kubernetes.Interface
+	eksapi                sdk.EKSAPI
+	previousVersion       fake.AtomicPtr[string]
```

Comment:
We should use the atomic pointer from the standard `sync/atomic` package, this is meant for our tests.